### PR TITLE
Add serialization support for dynamic textures

### PR DESCRIPTION
**Pull request by kcoley**
Fixes https://github.com/BabylonJS/Babylon.js/issues/4793

### CODE REVIEW

File: src/Materials/Textures/babylon.dynamicTexture.ts [hunk 1]
```diff
@@ -178,6 +178,20 @@
             return newTexture;
         }
 
+        /**
+         * Serializes the dynamic texture
+         * @returns a serialized dynamic texture object
+         */
+        public serialize(): any {
+            const serializationObject = super.serialize();
```
Comment (deltakosh) [inline src/Materials/Textures/babylon.dynamicTexture.ts]: Does it work on IE?

Comment (kcoley) [reply src/Materials/Textures/babylon.dynamicTexture.ts]: No, this does not work in IE.  I'll work on a fix.

Comment (deltakosh) [reply src/Materials/Textures/babylon.dynamicTexture.ts]: A fix or if none found a check to make sure it won't crash

**Comment by kcoley** [comment]
@deltakosh `toDataURL` works on IE11, but my playground was using a promise, which didn't work.  I changed it to use observable instead to test on IE11: http://playground.babylonjs.com/#FU0ES5

I also added a warning indicating that the scene should be ready before serializing the dynamic texture.  I can add this warning to the Dynamic Texture documentation as well.

**Comment by deltakosh** [comment]
Good!

**Pull request merged**